### PR TITLE
Allow AppSpecs to define a custom LockoutPanel component

### PR DESCRIPTION
**Pull request by TomTirapani**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: core/AppSpec.js [hunk 1]
```diff
@@ -40,11 +40,14 @@ export class AppSpec {
...
+     *      establish a connection and initiate a heartbeat.
      * @param {(Class|function)} [c.idlePanel] - optional custom Component to display when App has
-     *      been suspended.  The component will receive a single prop -- onReactivate -- a callback
+     *      been suspended. The component will receive a single prop -- onReactivate -- a callback
      *      called when the user has acknowledged the suspension and wishes to reload the app and
-     *      continue working.  Specify as a React Component or an element factory.
+     *      continue working. Specify as a React Component or an element factory.
+     * @param {(Class|function)} [c.lockoutPanel] - optional custom Component to display when the
+     *      user is denied access to app. Intended for apps that implement custom auth flows.
+     *      See also `lockoutMessage` for a more lightweight customization option.
```
Comment (amcclain) [inline core/AppSpec.js:50]: 👍 great - nice way to call out the two different options for customizing

**Review by amcclain — COMMENTED** [review]
Thanks for this - saw the PR on the CryptIQ side and happy to get it in, definitely could be useful.

Please see one note re. `elementFromContent` - let me know if there's any reason _not_ to use that?

**Comment by TomTirapani** [comment]
Thanks for taking a look! Re: `elementFromContent` - I didn't use it because we weren't modifying the props, but I agree that there's no reason not to use it and that it gives us a bit more flexibility. I've updated the branch.

**Comment by amcclain** [comment]
> Thanks for taking a look! Re: `elementFromContent` - I didn't use it because we weren't modifying the props, but I agree that there's no reason not to use it and that it gives us a bit more flexibility. I've updated the branch.

Great thanks - appreciate it.  I realize that the jsdoc for this and the preexisting `idlePanel` config should both likely be updated to remove the `Class` type - looking yet again at the react types to see if there is a better option I feel comfortable taking, but will merge this either way. 👍

**Review by amcclain — APPROVED** [review]

**Pull request merged**